test/mpi/rma/win_shared_query.c

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (C) by Argonne National Laboratory
+ *     See COPYRIGHT in top-level directory
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <assert.h>
+#include <mpi.h>
+#include "mpitest.h"
+
+#define ELEM_PER_PROC 100
+#define ELEM_SIZE sizeof(int)
+
+const int verbose = 0;
+
+int main(int argc, char **argv)
+{
+    int errors = 0;
+    MPI_Comm comm;
+    MPI_Win win;
+
+    MTest_Init(&argc, &argv);
+
+    comm = MPI_COMM_WORLD;
+
+    int rank, nproc;
+    MPI_Comm_rank(comm, &rank);
+    MPI_Comm_size(comm, &nproc);
+
+    int *my_base;
+    MPI_Win_allocate(ELEM_PER_PROC * ELEM_SIZE, ELEM_SIZE, MPI_INFO_NULL, comm, &my_base, &win);
```

Comment:
The commit message for this test says it checks a window with `MPI_Win_create` but the code only checks `MPI_Win_allocate`. We should add another check or amend the commit message.

Reply:
Good point. I enhanced the test to test `MPI_Win_create`, `MPI_Win_allocate`, and `MPI_Win_allocate_shared` based on the command line option

Reply:
The `MPI_Win_create` test reveals a miss for the case when `MPIDIG_WIN(win, shared_table)` does not exist. Fixup commit added.

Reply:
If we set `MPITEST_VERBOSE=1`
```
[0] TEST MPI_Win_create
[1] TEST MPI_Win_create
[2] TEST MPI_Win_create
[3] TEST MPI_Win_create
[0] [0] query [0 / 4]: 0x557f69896800 - size=400, disp_unit=4
[0] [0] query [1 / 4]: (nil) - size=0, disp_unit=0
[0] [0] query [2 / 4]: (nil) - size=0, disp_unit=0
[0] [0] query [3 / 4]: (nil) - size=0, disp_unit=0
[1] [1] query [0 / 4]: (nil) - size=0, disp_unit=0
[1] [1] query [1 / 4]: 0x55c6e8e19800 - size=400, disp_unit=4
[1] [1] query [2 / 4]: (nil) - size=0, disp_unit=0
[1] [1] query [3 / 4]: (nil) - size=0, disp_unit=0
[2] [2] query [0 / 4]: (nil) - size=0, disp_unit=0
[2] [2] query [1 / 4]: (nil) - size=0, disp_unit=0
[2] [2] query [2 / 4]: 0x55f150312520 - size=400, disp_unit=4
[2] [2] query [3 / 4]: (nil) - size=0, disp_unit=0
[3] [3] query [0 / 4]: (nil) - size=0, disp_unit=0
[3] [3] query [1 / 4]: (nil) - size=0, disp_unit=0
[3] [3] query [2 / 4]: (nil) - size=0, disp_unit=0
[3] [3] query [3 / 4]: 0x55c4155423b0 - size=400, disp_unit=4
[0]  No Errors
[0] TEST MPI_Win_allocate
[1] TEST MPI_Win_allocate
[2] TEST MPI_Win_allocate
[3] TEST MPI_Win_allocate
[0] [0] query [0 / 4]: 0x7fae08aa5000 - size=400, disp_unit=4
[0] [0] query [1 / 4]: (nil) - size=0, disp_unit=0
[0] [0] query [2 / 4]: 0x7fae08aa5190 - size=400, disp_unit=4
[0] [0] query [3 / 4]: (nil) - size=0, disp_unit=0
[1] [1] query [0 / 4]: (nil) - size=0, disp_unit=0
[1] [1] query [1 / 4]: 0x7f940e590000 - size=400, disp_unit=4
[1] [1] query [2 / 4]: (nil) - size=0, disp_unit=0
[1] [1] query [3 / 4]: 0x7f940e590190 - size=400, disp_unit=4
[2] [2] query [0 / 4]: 0x7f5dc4fb6000 - size=400, disp_unit=4
[2] [2] query [1 / 4]: (nil) - size=0, disp_unit=0
[2] [2] query [2 / 4]: 0x7f5dc4fb6190 - size=400, disp_unit=4
[2] [2] query [3 / 4]: (nil) - size=0, disp_unit=0
[3] [3] query [0 / 4]: (nil) - size=0, disp_unit=0
[3] [3] query [1 / 4]: 0x7fb5c1baa000 - size=400, disp_unit=4
[3] [3] query [2 / 4]: (nil) - size=0, disp_unit=0
[3] [3] query [3 / 4]: 0x7fb5c1baa190 - size=400, disp_unit=4
[0]  No Errors
[0] TEST MPI_Win_allocate_shared
[1] TEST MPI_Win_allocate_shared
[3] TEST MPI_Win_allocate_shared
[2] TEST MPI_Win_allocate_shared
[1] [0] query [0 / 2]: 0x7f2ccae68000 - size=400, disp_unit=4
[1] [0] query [1 / 2]: 0x7f2ccae68190 - size=400, disp_unit=4
[3] [1] query [0 / 2]: 0x7f7ad60be000 - size=400, disp_unit=4
[3] [1] query [1 / 2]: 0x7f7ad60be190 - size=400, disp_unit=4
[0] [0] query [0 / 2]: 0x7f236bc55000 - size=400, disp_unit=4
[0] [0] query [1 / 2]: 0x7f236bc55190 - size=400, disp_unit=4
[2] [1] query [0 / 2]: 0x7fd5f186f000 - size=400, disp_unit=4
[2] [1] query [1 / 2]: 0x7fd5f186f190 - size=400, disp_unit=4
[0]  No Errors
```